src/main/java/com/denizenscript/denizencore/scripts/commands/core/MongoCommand.java

```diff
@@ -0,0 +1,442 @@
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("command") MapTag command,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("find") MapTag findQuery,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("by_id") String findByID,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("insert") MapTag insert,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("update") MapTag oldData,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("new") MapTag newData,
+                                   @ArgName("upsert") boolean upsert,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("use_database") String newDatabase,
+                                   @ArgPrefixed @ArgDefaultNull @ArgName("use_collection") String newCollection) {
+        if (!CoreConfiguration.allowMongo) {
+            Debug.echoError(scriptEntry, "Mongo disabled by config!");
+            return;
+        }
+        String connectionId = CoreUtilities.toLowerCase(id);
+        Connection connection = mongoConnections.get(connectionId);
+        Runnable runnable = () -> {
+            try {
+                if (uri != null) {
+                    if (mongoConnections.containsKey(connectionId)) {
+                        Debug.echoError(scriptEntry, "Already connected to a server with ID '" + id + "'!");
+                        scriptEntry.setFinished(true);
+                        return;
+                    }
+                    if (database == null) {
+                        Debug.echoError(scriptEntry, "You must specify a Database!");
+                        scriptEntry.setFinished(true);
+                        return;
+                    }
+                    if (collection == null) {
+                        Debug.echoError(scriptEntry, "You must specify a Collection!");
+                        scriptEntry.setFinished(true);
+                        return;
+                    }
+                    DenizenCore.schedule(new AsyncSchedulable(new OneTimeSchedulable(() -> {
+                        String conStr = uri.getValue();
+                        if (!conStr.startsWith("mongodb://") && !conStr.startsWith("mongodb+srv://")) {
+                            conStr = "mongodb://" + conStr;
+                        }
+                        Debug.echoDebug(scriptEntry, "Connecting to Mongo server...");
+                        MongoClient con = null;
+                        MongoDatabase db = null;
+                        MongoCollection<Document> col = null;
+                        try {
+                            con = MongoClients.create(conStr);
+                            db = con.getDatabase(database);
+                            col = db.getCollection(collection);
+                        }
+                        catch (final Exception e) {

```

Comment:
this exception block never `return;`s, so it'll throw an error and then immediately follow with "connection did not error"